### ISSUE TITLE
Conversion Of External Mods Remaining Items

### ISSUE THREAD

**Issue by Cernelius** [triplea-game/triplea#1271]
There are the following 4 collections of external mods that are still referring to their own folders, while not having the required assets (5 of such repositories have been changed, but the following 4 are still broken):

https://github.com/triplea-maps/ww2v3_variants
https://github.com/triplea-maps/world_at_war_pbem
https://github.com/triplea-maps/ultimate_world_variants
https://github.com/triplea-maps/smalls_1939

Regarding "world_at_war_pbem", @ron-murhammer actually made it obsolete with the recent waw updates, thus it should be either updated or the pasted assets should be the ones of:
https://github.com/triplea-maps/world_at_war/releases/tag/0.3

I've tested that the first release of the mod is functional for 1.8.0.9.

Also, I want to point out there are two pointless repositories (I guess they should be deleted):
https://github.com/triplea-maps/new_world_order_variants-
https://github.com/triplea-maps/world_at_war_pbem-

I believe restoring or reworking the functionality of ww2v3_variants is important, also because it is currently available in download list.

**Comment by Cernelius** [comment]
It is not clear what to do with world_at_war_pbem, since importing the latest assets or referring to the original folder would not sane it anyway.

So, while pondering it, this reduces the folders that need to be put back working (by adding the original assets, since this is what the lead developers decided to do) to only 3:
https://github.com/triplea-maps/ww2v3_variants
https://github.com/triplea-maps/ultimate_world_variants
https://github.com/triplea-maps/smalls_1939

**Comment by DanVanAtta** [comment]
@Cernelius  a few questions if you would help clarify a few points:

> Regarding "world_at_war_pbem", @ron-murhammer actually made it obsolete with the recent waw updates, thus it should be either updated or the pasted assets should be the ones of:
https://github.com/triplea-maps/world_at_war/releases/tag/0.3

If obsolete, should it be deleted?

> Also, I want to point out there are two pointless repositories 

Why/how so pointless? I simply don't know. it does sound like they should be deleted, but I'd like to understand why before doing so.

> I believe restoring or reworking the functionality of ww2v3_variants is important, also because it is currently available in download list.

I agree, is it the only one broken that is also in the downloads list? Everything in the download list working is a priority.


------

Otherwise, for the variant maps that are not working, but are not in the download list: if they are worth fixing, are they also worth adding to the download list?

**Comment by Cernelius** [comment]
> > Regarding "world_at_war_pbem", @ron-murhammer actually made it obsolete with the recent waw updates, thus it should be either updated or the pasted assets should be the ones of:
> > https://github.com/triplea-maps/world_at_war/releases/tag/0.3
> 
> If obsolete, should it be deleted?

Rather than "obsolete" I should have said "outdated".
I don't know. It is a mod for "World_At_War", that worked before 2.0, but was broken by the not backwards compatible update of 2.0 thereafter. I've never played it and I've never seen anyone playing or hosting it.
Updating it would be a bit of work and possible to get something wrong, especially not knowing it. Best would be that the maker updates it and then tests that it works fine, in case. I'd let @ron-murhammer decide on this.
Since the "World_At_War" changes involve gameplay elements, no mods can be strictly updated, but they are going to be effected on a gameplay basis.
My suggestion is (and was) flagging this mod as obsolete and letting the owner or anyone interested updating it, in case; or the owner or @ron-murhammer should decide to delete it.
In the meantime, I suggest converting the "xml" into a regular "txt" (to disable it).
I guess deleting it would be the most sensible choice, as the mapmaker may redo something similar starting from the new World_At_War, if he wants to; but being someone else's work, I don't feel like suggesting it.

I suggest removing this item from this issue and redirect to:
https://github.com/triplea-game/triplea/issues/1272

Generally speaking, I suggest that any time an original skin is updated and breaks any games, those games should be either updated or kept as "txt", somehow flagging them as outdated.

> > Also, I want to point out there are two pointless repositories
> 
> Why/how so pointless? I simply don't know. it does sound like they should be deleted, but I'd like to understand why before doing so.

They are empty and are wrongly named, as having a "-" followed by nothing. I suggest deleting them.

> > I believe restoring or reworking the functionality of ww2v3_variants is important, also because it is currently available in download list.
> 
> I agree, is it the only one broken that is also in the downloads list? Everything in the download list working is a priority.

I believe it is.

> Otherwise, for the variant maps that are not working, but are not in the download list: if they are worth fixing, are they also worth adding to the download list?

I don't know, as I've never played any of them. I don't even know if they worked.

---

So, the items left for this issue are only:

Folders that need to be put back working (by adding the original assets, since this is what the lead developers decided to do):
https://github.com/triplea-maps/ww2v3_variants
https://github.com/triplea-maps/ultimate_world_variants
https://github.com/triplea-maps/smalls_1939

Folders that need to be deleted:
https://github.com/triplea-maps/new_world_order_variants-
https://github.com/triplea-maps/world_at_war_pbem-

**Comment by Cernelius** [comment]
It seems that:
https://github.com/triplea-maps/smalls_1939/blob/master/map/games/Smalls_1939.xml
is just the old (3.4) version of (3.6.6):
https://github.com/triplea-maps/big_world_variations/blob/master/map/games/Smalls_1939.xml
with the info name changed from "Small's 1939" to "Big World : Small's 1939".

If it is so, then I change my full suggestions list to:

Folders that need to be put back working (by adding the original assets, since this is what the lead developers decided to do):
https://github.com/triplea-maps/ww2v3_variants
https://github.com/triplea-maps/ultimate_world_variants

Folders that need to be deleted:
https://github.com/triplea-maps/smalls_1939
https://github.com/triplea-maps/new_world_order_variants-
https://github.com/triplea-maps/world_at_war_pbem-

**Comment by Cernelius** [comment]
I've started some of the ultimate_world_variants mods in 1.8.0.9 and they seem playable. It seems that one of them, namely Ultimate_World_-_Expanded.xml, was made by the original mapmaker.

**Comment by Cernelius** [comment]
It appears that the required assets haven't been successfully pasted into "big_world_variations", and it is still a broken collection of external mods:
https://github.com/triplea-maps/big_world_variations

> Loading map: big_world_variations, from: C:\Users\Totale\triplea\downloadedMaps\big_world_variations-master.zipFailed to launch
> 
> Loading resources from the following paths: [C:\Users\Totale\triplea\downloadedMaps\big_world_variations-master.zip, C:\Users\Totale\triplea\games\TripleA\triplea-1.9.0.0.3266-all_platforms\assets]
> Loading map: big_world_variations, from: C:\Users\Totale\triplea\downloadedMaps\big_world_variations-master.zip
> Loading resources from the following paths: [C:\Users\Totale\triplea\downloadedMaps\big_world_variations-master.zip, C:\Users\Totale\triplea\games\TripleA\triplea-1.9.0.0.3266-all_platforms\assets]
> java.lang.reflect.InvocationTargetException
> 	at java.awt.EventQueue.invokeAndWait(Unknown Source)
> 	at java.awt.EventQueue.invokeAndWait(Unknown Source)
> 	at javax.swing.SwingUtilities.invokeAndWait(Unknown Source)
> 	at games.strategy.triplea.TripleA.startGame(TripleA.java:122)
> 	at games.strategy.engine.framework.startup.launcher.ServerLauncher.launchInNewThread(ServerLauncher.java:167)
> 	at games.strategy.engine.framework.startup.launcher.AbstractLauncher.lambda$launch$116(AbstractLauncher.java:43)
> 	at games.strategy.engine.framework.startup.launcher.AbstractLauncher$$Lambda$119/1758954757.run(Unknown Source)
> 	at java.lang.Thread.run(Unknown Source)
> Caused by: java.lang.IllegalStateException: Error in resource loading. Unable to load expected resource: polygons.txt, the error is that either we did not find the correct path to load. Check the resource loader to make sure the map zip or dir was added. Failing that, the path in this error message should be available relative to the map folder, or relative to the root of the map zip
> 	at games.strategy.triplea.ui.mapdata.MapData.<init>(MapData.java:155)
> 	at games.strategy.triplea.ui.UIContext.internalSetMapDir(UIContext.java:77)
> 	at games.strategy.triplea.ui.AbstractUIContext.setDefaultMapDir(AbstractUIContext.java:118)
> 	at games.strategy.triplea.ui.TripleAFrame.<init>(TripleAFrame.java:227)
> 	at games.strategy.triplea.TripleA.lambda$startGame$335(TripleA.java:124)
> 	at games.strategy.triplea.TripleA$$Lambda$126/1815487610.run(Unknown Source)
> 	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
> 	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
> 	at java.awt.EventQueue.access$500(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.awt.EventQueue$3.run(Unknown Source)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
> 	at java.awt.EventQueue.dispatchEvent(Unknown Source)
> 	at games.strategy.engine.framework.GameRunner$1.dispatchEvent(GameRunner.java:360)
> 	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
> 	at java.awt.EventDispatchThread.run(Unknown Source)

Lobby as of today:

> (00:20:56) Hummel:  hm. what was that? :-)
> (00:21:00) Goreg:  can you try coming to my host?
> (00:21:02) Goreg:  I have no diea
> (00:21:05) Hummel:  sure
> FloppyHat has left
> (00:21:46) n0m3:  anyone for revised?
> (00:21:48) n0m3:  bot 2
> (00:22:06) Hummel:  hm
> (00:22:14) Hummel:  there seems to be an error
> (00:22:26) Goreg:  "problem during startup, game aborted"
> (00:22:32) Goreg:  try a bot?
> (00:22:32) Hummel:  has anyone started and played a big world small's map in this new version yet?
> (00:22:49) Hummel:  sure
> (00:22:56) Hummel:  2?
> (00:23:04) Goreg:  did you download the map?
> (00:23:08) Hummel:  yes
> (00:23:11) Goreg:  ok
> (00:23:38) Goreg:  yeah 2
> majorQ10 has joined
> yippi has joined
> (00:24:44) Hummel:  hm
> (00:24:47) Goreg:  damn that's irritating
> (00:24:47) Hummel:  i'll restart
> (00:24:50) Goreg:  ok me too
> Hummel has left
> yippi has left
> Goreg has left
> Goreg has joined
> Hummel has joined
> (00:25:34) Goreg:  damn, doesn't even work when i try by myself
> (00:25:39) Goreg:  You were slapped by Goreg
> (00:25:39) n0m3:  anyone for revised?
> (00:25:41) n0m3:  bot 2?
> (00:25:51) Goreg:  Cernel, do you know why big world isn't working?
> (00:26:29) Hummel:  Unable to load expected resource: polygons.txt, the error is that either we did not find the correct path to load.
> (00:26:46) Hummel:   Check the resource loader to make sure the map zip or dir was added. 
> (00:26:50) Hummel:  hm
> (00:26:53) Hummel:  but i did. i think
> Firesmith has left
> (00:28:10) Goreg:  do you want to use the old version of triplea?
> (00:29:14) Trojan:  i can be rdy in 10 minutes
> (00:29:24) Trojan:  but save we play in old lobby just
> Firesmith has joined
> (00:29:43) Goreg:  try old lobby?
> (00:29:44) Hummel:  yes, just checked in engine settings
> (00:30:01) Trojan:  saves not work here btw
> (00:30:01) Hummel:  sure
> (00:30:04) Trojan:  old saves
> (00:30:06) Cernel:  big world isn't working???
> (00:30:12) Trojan:  saves old
> (00:30:13) Goreg:  yeah not online
> (00:30:14) Trojan:  listen
> (00:30:15) Goreg:  it seems
> Damon53 has left
> FloppyHat has joined
> (00:30:29) Cernel:  join my host
> Maeert has left
> Damon53 has joined
> (00:30:46) Trojan:  nOm3 or Khan
> (00:30:49) Trojan:  conti?
> FloppyHat has left
> (00:31:10) TheKhan:  hi in game
> Hummel has left
> (00:31:15) Trojan:  gl
> Kaptain has left
> (00:32:58) Trojan:  game anyone ?
> (00:34:20) Goreg:  Cernel, it seems that Big World is working, but Big World 1939: Small's is not working

Moreover, I don't know what the following are for:
https://github.com/triplea-maps/big_world_variations/blob/master/map/NekahNets_map.properties
https://github.com/triplea-maps/big_world_variations/blob/master/map/PrussiaJune_map.properties
https://github.com/triplea-maps/big_world_variations/blob/master/map/Smalls_map.properties

Were mods able to use them, while referring to the other assets at the same time?

**Comment by Cernelius** [comment]
Updated "to-do" list:

Folders that need to be put back working (by adding the original assets, since this is what the lead developers decided to do):
https://github.com/triplea-maps/big_world_variations
https://github.com/triplea-maps/ww2v3_variants
https://github.com/triplea-maps/ultimate_world_variants

Folders that need to be deleted:
https://github.com/triplea-maps/smalls_1939
https://github.com/triplea-maps/new_world_order_variants-
https://github.com/triplea-maps/world_at_war_pbem-

**Comment by Cernelius** [comment]
> (03:32:54) blinchik:  hey cernel
> (03:32:59) blinchik:  it seems nekahnet is not running on here
> (03:33:05) Cernel:  yup yup
> (03:33:06) blinchik:  cause ijust loaded bw v3 no problem
> (03:33:09) blinchik:  but nekah keeps crashing on me
> (03:33:12) blinchik:  oh.. you knew?
> (03:33:13) prastle:  in experimental dl
> (03:33:15) Cernel:  yeah
> (03:33:27) prastle:  or that doesnt run?
> (03:33:33) Cernel:  nekanet is currently unplayable
> (03:33:41) prastle:  Oo
> (03:33:41) Elenol:  ahahahhaa
> (03:33:46) Elenol:  poor blinch

**Comment by ron-murhammer** [comment]
@Cernelius Yeah it appears you are correct that a number of the variants are broken.

**Comment by Cernelius** [comment]
It remains only:

Folders that need to be put back working (by adding the original assets, since this is what the lead developers decided to do):
https://github.com/triplea-maps/ultimate_world_variants

Which I believe it is low priority, as it is not currently available in download list. I tested some of the mods inside and they worked with 1.8.0.9. I don't know if it should be added to download list or not. I believe Ultimate_World_-_Expanded.xml was made by the original mapmaker (strange it is not part of the original map); the Developer Forum tread for this mod is:
http://tripleadev.1671093.n2.nabble.com/ULTIMATE-WORLD-EXPANDED-tp3506774.html

I verified all other items appear to have been correctly addressed.

Since all cases belonging to download list have been solved, I think this issue can be also closed by just noting the remaining item in:
https://github.com/triplea-game/triplea/wiki/Broken-Maps
